### PR TITLE
add pytest preps via ruff

### PR DESCRIPTION
**Pull request by ahms5**
- add support of pytest preps
- list can be find here https://docs.astral.sh/ruff/rules/#flake8-pytest-style-pt
- i ignored some, please see lint.ignore

### CODE REVIEW

**Review by f-brinkmann — APPROVED** [review]
The only thing you actively changes is *pyproject.toml*, right? I would be fine with enforcing the changes caused by that. It looks a little cleaner to me in most cases.

**Comment by ahms5** [comment]
basically correct, but I had to fix some ruff warnings by hand.

File: pyproject.toml [hunk 1]
```diff
@@ -5,10 +5,15 @@ exclude = [
     "setup.py",
 ]
 line-length = 79
-lint.ignore = []
+lint.ignore = [
+    "PT011", # use match in pytest.raises
```
Comment (pingelit) [inline pyproject.toml:9]: Why not include this?

**Review by pingelit — DISMISSED** [review]
Thanks for adding this, I would also add `PT011` but this is probably another big chink of work.

File: pyproject.toml (hunk 1 above)
Comment (ahms5) [reply pyproject.toml:9]: Surely it would make sense. I think I was just too lazy to add it manually at any time.

**Comment by ahms5** [comment]
let's wait for the discussion in https://github.com/pyfar/pyfar/issues/647

**Review by f-brinkmann — DISMISSED** [review]

**Comment by ahms5** [comment]
closes in favor of #692

**Pull request closed without merging**